osmosis-labs/osmosis-frontend
# Use data services for CL balances, add initial tick/ current tick buffer

## What is the purpose of the change



When the router first loaded, the CL pool amounts weren't loaded and we're 0, causing the router to not route through CL pools (they didn't appear to have liquidity) _when the page first loads._ Now, I added a general feature to add optional "metrics" data to the query pool object that can be used directly instead of queried in the case of CL. With our filtered-pools query from Imperator, we're able to use the token amounts (for CL) as well as other liquidity and volume data that's included. This prevents us from having to query all of the CL pools. Further, it will fall back to the individual queries if the optimized pools query is not available.

Also, I added a buffer to our initial tick estimation logic. This prevents us from querying behind the current tick (invalid) since the current tick can move as prices are always shifting.

Also, consequently it fixes Kava USDT not appearing in dropdowns.

## Brief Changelog



* Add useRoutablePoolsHook, containing query await logic
* Fixes CL pools being included in initial load
* Removes some unnecessary node version queries

## Testing and Verifying



* Notice on refresh, CL pools (and their assets) are included in swap router visualization
* Notice no tick query errors in network tab of dev tools

## Documentation and Release Note
